File: src/main/java/com/systemservices/kata/service/EmployeeService.java
package com.systemservices.kata.service;

import com.systemservices.kata.model.Employee;
import com.systemservices.kata.repository.EmployeeRepository;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class EmployeeService {

  private final EmployeeRepository repo;

  public EmployeeService(EmployeeRepository repo) {
    this.repo = repo;
  }

  public List<Employee> findAll() {
    return repo.findAll();
  }

  public Optional<Employee> findById(Long id) {
    return repo.findById(id);
  }

  public Employee create(Employee e) {
    return repo.save(e);
  }

  public Employee update(Long id, Employee e) {
    e.setId(id);
    return repo.save(e);
  }

  public void delete(Long id) {
    repo.deleteById(id);
  }

  public boolean exists(Long id) {
    return repo.existsById(id);
  }
}

File: openspec-py/app/service/employee_service.py
from typing import Optional

from app.model.employee import Employee
from app.repository.employee_repository import EmployeeRepository


class EmployeeService:
    def __init__(self, repository: EmployeeRepository) -> None:
        self._repository = repository

    def find_all(self) -> list[Employee]:
        return self._repository.find_all()

    def find_by_id(self, id: int) -> Optional[Employee]:
        return self._repository.find_by_id(id)

    def create(self, employee: Employee) -> Employee:
        return self._repository.save(employee)

    def update(self, id: int, employee: Employee) -> Employee:
        employee.id = id
        return self._repository.save(employee)

    def delete(self, id: int) -> None:
        self._repository.delete_by_id(id)

    def exists(self, id: int) -> bool:
        return self._repository.exists_by_id(id)

